lib/associations/mixin.js

```diff
@@ -209,18 +209,27 @@ const Mixin = {
     return association;
   },
 
-  getAssociation(target, alias) {
-    for (const associationName in this.associations) {
-      if (this.associations.hasOwnProperty(associationName)) {
-        const association = this.associations[associationName];
+  getAssociations(target) {
+    return Object.values(this.associations).filter(association => association.target.name === target.name);
+  },
 
-        if (association.target.name === target.name && (alias === undefined ? !association.isAliased : association.as === alias)) {
-          return association;
-        }
-      }
+  getAssociationForAlias(target, alias) {
+    // Two associations cannot have the same alias, so we can use find instead of filter
+    const association = this.getAssociations(target)
+      .find(association => this.verifyAssociationAlias(association, alias));
+    if (!association) {
+      return null;
```

Comment:
This is redundant, `find()` already returns `undefined` when the item was not found